src/engine/source/metrics/include/metrics/dataHubExporter.hpp

```diff
@@ -0,0 +1,87 @@
+#ifndef __JSON_EXPORTER_H
+#define __JSON_EXPORTER_H
+
+#include "opentelemetry/common/spin_lock_mutex.h"
+#include "opentelemetry/sdk/metrics/data/metric_data.h"
+#include "opentelemetry/sdk/metrics/instruments.h"
+#include "opentelemetry/sdk/metrics/push_metric_exporter.h"
+#include "opentelemetry/version.h"
+#include <iostream>
+#include <json/json.hpp>
+#include <string>
+
+#include <metrics/iDataHub.hpp>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter
+{
+namespace metrics
+{
+
+/**
+ * The DataHubExporter exports record data through an ostream
+ */
+class DataHubExporter final : public opentelemetry::sdk::metrics::PushMetricExporter
+{
+public:
+    /**
+     * Create an DataHubExporter. This constructor takes in a reference to an ostream that
+     * the export() function will send metrics data into. The default ostream is set to
+     * stdout
+     */
+    explicit DataHubExporter(std::shared_ptr<metrics_manager::IDataHub> dataHub,
+                          sdk::metrics::AggregationTemporality aggregation_temporality =
+                              sdk::metrics::AggregationTemporality::kCumulative) noexcept;
+
+    /**
+     * Export
+     * @param data metrics data
+     */
```

Comment:
Use copydoc In all the methods to be overwrited
https://doxygen.nl/manual/commands.html#cmdcopydoc